### PR TITLE
Remove special string node type

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
It occurred to me in the meeting today that we don't actually need/want a special string string literal expression.  Rather the singel string expression could have multiple kinds of tokens underneath it, all representing a constant string literal.  

Relates to test plan https://github.com/dotnet/roslyn/issues/55306

### CODE REVIEW

File: src/Compilers/CSharp/Portable/PublicAPI.Unshipped.txt [hunk 1]
```diff
@@ -33,9 +33,7 @@ Microsoft.CodeAnalysis.CSharp.SyntaxKind.InterpolatedMultiLineRawStringEndToken
 Microsoft.CodeAnalysis.CSharp.SyntaxKind.InterpolatedMultiLineRawStringStartToken = 9082 -> Microsoft.CodeAnalysis.CSharp.SyntaxKind
 Microsoft.CodeAnalysis.CSharp.SyntaxKind.InterpolatedSingleLineRawStringEndToken = 9081 -> Microsoft.CodeAnalysis.CSharp.SyntaxKind
 Microsoft.CodeAnalysis.CSharp.SyntaxKind.InterpolatedSingleLineRawStringStartToken = 9080 -> Microsoft.CodeAnalysis.CSharp.SyntaxKind
-Microsoft.CodeAnalysis.CSharp.SyntaxKind.MultiLineRawStringLiteralExpression = 9075 -> Microsoft.CodeAnalysis.CSharp.SyntaxKind
```
Comment (jcouv) [inline src/Compilers/CSharp/Portable/PublicAPI.Unshipped.txt:36]: Consider resolving PROTOTYPE marker and updating compiler test plan while you're making a change

Comment (jcouv) [reply src/Compilers/CSharp/Portable/PublicAPI.Unshipped.txt:36]: Never mind, will probably need another PR anyways. I updated the test plan based on yesterday's feature review:
[https://github.com/dotnet/roslyn/issues/55306](https://github.com/dotnet/roslyn/issues/55306)

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 8)

**Comment by jcouv** [comment]
@chsienki @dotnet/roslyn-compiler for second review (relates to raw string literals). Thanks

**Comment by cston** [comment]
Has the public API gone through API review?

**Comment by jcouv** [comment]
We discussed API review during the feature review. Cyrus is going to start the process.

File: src/Compilers/CSharp/Portable/Syntax/LiteralExpressionSyntax.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    public partial class SyntaxFactory
+    {
+        /// <summary>Creates a new LiteralExpressionSyntax instance.</summary>
+        public static LiteralExpressionSyntax LiteralExpression(SyntaxKind kind)
+            => SyntaxFactory.LiteralExpression(kind, SyntaxFactory.Token(GetLiteralExpressionTokenKind(kind)));
+
+        private static SyntaxKind GetLiteralExpressionTokenKind(SyntaxKind kind)
```
Comment (cston) [inline src/Compilers/CSharp/Portable/Syntax/LiteralExpressionSyntax.cs:16]: >GetLiteralExpressionTokenKind [](http://example.com/codeflow?start=34&length=29)

Consider asserting: 
```csharp
var result = kind switch { ... };
Debug.Assert(kind == SyntaxKindFacts.GetLiteralExpression(result));
return result;
```

File: src/Compilers/CSharp/Portable/Syntax/LiteralExpressionSyntax.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+using System;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    public partial class SyntaxFactory
+    {
+        /// <summary>Creates a new LiteralExpressionSyntax instance.</summary>
+        public static LiteralExpressionSyntax LiteralExpression(SyntaxKind kind)
```
Comment (RikkiGibson) [inline src/Compilers/CSharp/Portable/Syntax/LiteralExpressionSyntax.cs:13]: I didn't understand why these methods moved out of the generated file. Could you please clarify?

**Review by RikkiGibson — APPROVED** [review]
LGTM aside from one question

File: src/Compilers/CSharp/Portable/Syntax/LiteralExpressionSyntax.cs (hunk 3 above)
Comment (CyrusNajmabadi) [reply src/Compilers/CSharp/Portable/Syntax/LiteralExpressionSyntax.cs:13]: they're not getting generated autoamtically as there isn't a 1:1 correspondance with node-type and node-kind.  note: we will discuss this at the API review.  I personally think we don't need/want the 1:1 correspondence.

**Comment by jcouv** [comment]
@cston The issue tracking public API review is here: https://github.com/dotnet/roslyn/issues/59046

**Pull request merged**